lib/services/delete_item_from_warehouse.rb

```diff
@@ -0,0 +1,18 @@
+module Shop
+  class DeleteItemFromWarehouse
+    def call(id, product_id)
+      warehouse_item = WAREHOUSE.find do |warehouse|
+        warehouse.id == id
+        warehouse.product_id == product_id
```

Comment:
you cannot compare 2 different stuff, ruby will ignore first, if you want to compare 2 different stuff you need to join them with some logic: && or ||